internal/pkg/skuba/replica/testing.go

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2019 SUSE LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package replica
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var m1 = createNode("master1")
+var m2 = createNode("master2")
+var m3 = createNode("master3")
+var m4 = createNode("master4")
+var m5 = createNode("master5")
+
+func createNode(name string) corev1.Node {
+	return corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: metav1.NamespaceSystem,
+			Labels:    map[string]string{"node-role.kubernetes.io/master": ""},
+		},
+	}
+}
+
+var d1R2 = createDeployment("deployment1", 2)
+var d1R3 = createDeployment("deployment2", 3)
+
+func createDeployment(deployment string, replica int) appsv1.Deployment {
+	return appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment,
+			Namespace: metav1.NamespaceSystem,
+			Labels:    map[string]string{"caasp.suse.com/skuba-replica-ha": "true"},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: func() *int32 { i := int32(replica); return &i }(),
```

Comment:
if change input parameter replica to int32, would this become easier? like `Replicas: &replica` [2]

Reply:
[3]